### PR TITLE
Improve LST workflows and move custom function to `std` one

### PR DESCRIPTION
**Pull request by VourMa**
In this PR I have:
- Moved the LST workflows to use the latest default geometry (D110)
- Made the `0.704` run on a GPU only if there is one, otherwise fall back to CPU.
- Replaced the custom binary search function with `std::lower_bound`.
- Made a couple more minor fixes that need to be done (e.g. using `MessageLogger` instead of `printf`, renamings, etc.)
- Got rid of the warning about the `SeedStopInfo` size mismatch between the `TrackCandidate` and the `TrajectorySeed` collections.

### CODE REVIEW

**Comment by VourMa** [comment]
@slava77 I cannot make LST run with the `29834 ` workflow, even though it is explicitly added. It seems like the `0.703` and `0.704` offsets refuse to be added to PU workflows. Any obvious reason why?

**Comment by slava77** [comment]
> @slava77 I cannot make LST run with the `29834 ` workflow, even though it is explicitly added. It seems like the `0.703` and `0.704` offsets refuse to be added to PU workflows. Any obvious reason why?

`UpgradeWorkflowTracking` from which `UpgradeWorkflow_lstOnCPUIters01TrackingOnly` inherit has `and not 'PU' in key`

**Comment by VourMa** [comment]
> UpgradeWorkflowTracking from which UpgradeWorkflow_lstOnCPUIters01TrackingOnly and other LST inherit has and not 'PU' in key

That's in the `condition` function and I though we were overriding it. But we need to define `condition()` and not `condition_()`  to do that... I have been failing to notice the little, sneaky underscore... Thanks!

File: Configuration/PyReleaseValidation/python/relval_gpu.py [hunk 1]
```diff
@@ -70,13 +70,13 @@
            13061.402,
 
            # Run4, Alpaka-based noPU
-           29634.402, 29634.403, 29634.404, 29634.406,
+           29634.402, 29634.403, 29634.404, 29634.406, 29634.704,
            29661.402,
            # Run4, Alpaka-based noPU GPU LST tracking D98
            24834.704,
```
Comment (slava77) [inline Configuration/PyReleaseValidation/python/relval_gpu.py:76]:
```suggestion
```
can remove, I guess

**Comment by ariostas** [comment]
Just out of curiosity, is it still worth having .703 when .704 can run on CPU? Is there a way to add a flag to run it on CPU even if there is a GPU?

File: Configuration/PyReleaseValidation/python/relval_gpu.py (hunk 1 above)
Comment (VourMa) [reply Configuration/PyReleaseValidation/python/relval_gpu.py:76]: Yes, I must have missed it.

**Comment by VourMa** [comment]
> Just out of curiosity, is it still worth having .703 when .704 can run on CPU? Is there a way to add a flag to run it on CPU even if there is a GPU?

For sure one can do that for the PR tests by asking for the test parameters below:
```
'--accelerators' : 'cpu'
```

I wasn't sure whether one can do it with `runTheMatrix.py` but I just checked and tested that:
```runTheMatrix.py -w upgrade -l 29634.704 --command "--accelerators cpu"```
and
```runTheMatrix.py -w upgrade -l 29634.704 --command "--accelerators gpu-nvidia"```
work as we want them to.

As a result, I think I can remove the the `.703` workflow (CPU-only) and rename `.704` (hardware-agnostic) to that. That would mean that our workflow would only live in the `relval_gpu.py` file and will be removed from `relval_Run4.py` (as it would cause issues in IB tests). I am not sure that this implies anything bad for us. @slava77 may know more or have a better idea whether we would like or wouldn't like to do that.

**Comment by slava77** [comment]
> As a result, I think I can remove the the `.703` workflow (CPU-only) and rename `.704` (hardware-agnostic) to that. That would mean that our workflow would only live in the `relval_gpu.py` file and will be removed from `relval_Run4.py`

I think we need to have LST running in GPU and CPU if possible in IBs. 

I'm not sure about other workflows in the _gpu.py file: are they forcing GPU or is it imposed at submission with `--accelerators` flag? If unclear, better ask

**Comment by VourMa** [comment]
> > As a result, I think I can remove the the `.703` workflow (CPU-only) and rename `.704` (hardware-agnostic) to that. That would mean that our workflow would only live in the `relval_gpu.py` file and will be removed from `relval_Run4.py`
> 
> I think we need to have LST running in GPU and CPU if possible in IBs.
> 
> I'm not sure about other workflows in the _gpu.py file: are they forcing GPU or is it imposed at submission with `--accelerators` flag? If unclear, better ask

In this case, let me proceed with the PR as is (will be after I add the minor fixes I mention in the description) and ask the question there - it's easier to remove rather than re-add.

**Review by slava77 — APPROVED** [review]

**Comment by VourMa** [comment]
Closing this in favor of #140 (same developments but with squashed commits)

**Pull request closed without merging**